pallets/proposals/src/impls.rs

```diff
@@ -172,6 +172,13 @@ impl<T: Config> Pallet<T> {
                 p.withdrawn_funds = p.withdrawn_funds.saturating_add(withdrawable);
                 if p.withdrawn_funds == p.raised_funds {
                     <T as Config>::DepositHandler::return_deposit(p.deposit_id)?;
+                    // TODO: reinstate storage deposit
+                    CompletedProjects::<T>::try_mutate(project_account, |completed_projects| -> DispatchResult {
+                        if !completed_projects.iter().any(|&completed_project| completed_project == project_key) {
+                                completed_projects.try_push(project_key).expect("The maximum number of projects has been reached for an account");
```

Comment:
We for sure should not panic here.
```suggestion
                                completed_projects.try_push(project_key).map_err(|_|, Error::<T>::TooManyProjects)?;
```